src/check.c

```diff
@@ -440,20 +443,18 @@ exec_check(int argc, char **argv)
 					rc = EXIT_FAILURE;
 				}
 				if (rc == EXIT_FAILURE)
-					goto cleanup;
+					break;
 				pkgdb_downgrade_lock(db, PKGDB_LOCK_EXCLUSIVE,
 				    PKGDB_LOCK_ADVISORY);
 			}
 			else {
 				rc = EXIT_FAILURE;
-				goto cleanup;
+				break;
 			}
 		}
-		pkgdb_it_free(it);
 		i++;
 	} while (i < argc);
 
-cleanup:
```

Comment:
Let's perhaps `assert(it == NULL);` here and use that as a canary that we may have accidentally leaked it somewhere; just need to additionally set it to NULL after the pkgdb_it_free() you added in the previous iteration up on line 361.